service/Payment.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen PrestaShop plugin
+ *
+ * Copyright (c) 2020 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ */
+
+namespace Adyen\PrestaShop\service;
+
+class Payment
+{
+    const AFTERPAY_PAYMENT_METHOD = 'afterpay';
+    const RATEPAY_PAYMENT_METHOD = 'ratepay';
+    const KLARNA_PAYMENT_METHOD = 'klarna';
+    const KLARNA_B2B_PAYMENT_METHOD = 'klarna_b2b';
+    const KLARNA_ACCOUNT_PAYMENT_METHOD = 'klarna_account';
+    const KLARNA_PAYNOW_PAYMENT_METHOD = 'klarna_paynow';
```

Comment:
as more and more is coming can't we regex on klarna ?

Reply:
It is possible yes although learning from the Magento2 plugin experiences I would keep a list of the full payment method names instead of using regexes because there are always exceptions.

A good example: https://github.com/Adyen/adyen-magento2/blob/78c4f4af3f5b183e363214007254c075e287da37/Gateway/Request/CheckoutDataBuilder.php#L131 we check if the payment method is open invoice pm and also avoid afterpay_touch. 

Another exmaple for the confusion in Magento2: https://github.com/Adyen/adyen-magento2/pull/615

It will be harder to maintain because every time we are going to have the new open invoice payment methods into the list but with this approach we can make sure that we are in control of the payment methods to be categorised open invoice methods.